docs/apis/streaming/windows.md

```diff
@@ -24,1023 +24,593 @@ specific language governing permissions and limitations
-</table>
-
+input
+    .keyBy(<key selector>)
+    .window(<window assigner>)
+    .<windowed transformation>(<window function>)
+{% endhighlight %}
 </div>
 </div>
 
-### Advanced Window Constructs
+We will cover the different window assigners in [window assigners](#window-assigners).
+
+The window transformation can be one of `reduce()`, `fold()` or `apply()`. Which respectively
+takes a `ReduceFunction`, `FoldFunction` or `WindowFunction`. We describe each of these ways
+of specifying a windowed transformation in detail in [window functions](#window-functions).
+
+For more advanced use cases you can also specify a `Trigger` that determines when exactly a window
+is being considered as *ready for processing*. These will be covered in more detail in
+[triggers](#triggers).
+
+## Window Assigners
+
+The window assigner specifies how elements of the stream are divided into slices. You can provide
+your custom window assigner by implementing `WindowAssigner` but Flink comes with
+window assigners for typical use cases: *tumbling windows*, *sliding windows*, *session windows*
+and *global windows*. Except for the last, all of these assign windows based on time, which
+can either be processing time or event time.
+
+Let's first look at how each of these window assigners works before looking at how they can be used
+in a Flink program. We will be using abstract figures to visualize the workings of each assigner:
+in the following, the purple circles are elements of the stream, they are partitioned
+by some key (in this case *user 1*, *user 2* and *user 3*) and the x-axis shows the progress
+of time.
+
+### Global Windows
+
+Global windows are a way of specifying that we don't want to subdivide our elements into windows.
+Each element is assigned to the one single *global window* (still separate for each key, of course).
+This is only useful if you also specify a custom [trigger](#triggers), otherwise you will never
+process any data because the global window does not have a natural end at which we could process
+the aggregated elements.
+
+<img src="non-windowed.svg" class="center" style="width: 80%;" />
 
-The general mechanism can define more powerful windows at the cost of more verbose syntax. For example,
-below is a window definition where windows hold elements of the last 5 seconds and slides every 1 second,
-but the execution of the window function is triggered when 100 elements have been added to the
-window, and every time execution is triggered, 10 elements are retained in the window:
+### Tumbling Windows
+
+A *tumbling windows* assigner assigns elements to fixed time buckets of a specified *window size*.
+For example, if you specify a window size of 5 minutes, the window function will get 5 minutes
+worth of elements in each invocation.
+
+<img src="tumbling-windows.svg" class="center" style="width: 80%;" />
+
+### Sliding Windows
+
+The *sliding windows* assigner is very similar to the *tumbling windows* assigner but it assigns
```

Comment:
The *sliding windows* assigner assigns elements to windows of fixed length equal to *window size*, as the *tumbling windows* one, but in this case, windows can be overlapping. The size of the overlap is defined by a user-specified parameter called the *window slide*. As windows are overlapping, an element can be assigned to multiple windows, as shown below.